File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] arr, int k) {
        int[] ans=new int[2];
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<arr.length;i++){
            if(map.containsKey(k-arr[i])){
                ans[0]=i;
                ans[1]=map.get(k-arr[i]);
                return ans;
            }
            map.put(arr[i],i);

        }

        return ans;
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, arr: List[int], k: int) -> List[int]:
        ans:List[int]=[]
        map:Dict[int,int]={}
        for i in range(len(arr)):
            complement =k-arr[i]
            if complement in map:
                ans=[i,map[complement]];
                return ans
            map[arr[i]]=i
        return ans